### PR TITLE
Allow commands to abort pipeline execution

### PR DESCRIPTION
**Pull request by kevsmith**
Implements Ruby command-facing API for the design described in [#691](https://github.com/operable/cog/issues/691).

Fixes [#691](https://github.com/operable/cog/issues/691)

### CODE REVIEW

File: lib/cog/response.rb [hunk 1]
```diff
@@ -5,18 +5,23 @@ class Cog
...
     end
 
-    def send
-      return if content.nil?
+    def abort
+      @aborted = true
+    end
 
+    def send
+      write "COGCMD_ACTION: stop" unless @aborted.nil?
```
Comment (christophermaier) [inline lib/cog/response.rb]: Maybe `COGCMD_ACTION: abort` for consistency?

Also, are there other values for `COGCMD_ACTION`?

Comment (christophermaier) [inline lib/cog/response.rb]: The value is "abort" in the Pycog and Relay PRs... guess this slipped through in the renaming.

**Comment by christophermaier** [comment]
If "abort" is for handled error scenarios but an error is unhandled, we should probably revise `Cog::Command#fail` and the rescue block from `Cog::Command#execute`. Current uses of `#fail` should migrate over to using `#abort` + appropriate templates (meaning `#fail` should be deprecated), while the rescue block should be loosened to support any exception.

File: lib/cog/response.rb (hunk 1 above)
Comment (kevsmith) [inline lib/cog/response.rb]: Whoops. This is an oversight. So many repos to keep in sync.....

**Comment by kevsmith** [comment]
Sounds reasonable for exception handling but out of scope for this initial pass, I think. I was focusing on getting the ability implemented in all the right places figuring then we can adapt command behavior to take advantage of it.

**Comment by christophermaier** [comment]
Sounds good. We've got a number of commands that use `#fail` right now, so it'll take a bit of typing to get everything migrated over anyway; that shouldn't block this.

**Pull request merged**